ui/src/api/documents/document.js

```diff
@@ -33,16 +38,75 @@ const viewEvent = async docPid => {
   });
 };
 
-const createRelation = async (docPid, data) => {
-  const resp = await http.post(`${documentURL}${docPid}/relations`, data);
+const createRelation = async (
+  referer,
+  selectedRelatedList,
+  relationType,
+  extraRelationField = {}
+) => {
+  let newRelationsPayload = [];
+  if (relationType === 'serial' || relationType === 'multipart_monograph') {
+    const payload = parentChildRelationPayload(
+      relationType,
+      extraRelationField,
+      selectedRelatedList[0],
+      referer
+    );
+    newRelationsPayload.push(payload);
+  } else {
+    if (relationType === 'other') {
+      newRelationsPayload.push(
+        siblingRelationPayload(
+          relationType,
+          extraRelationField,
+          selectedRelatedList[0]
+        )
+      );
+    } else {
+      selectedRelatedList.map(selection =>
+        newRelationsPayload.push(
+          siblingRelationPayload(relationType, extraRelationField, selection)
+        )
+      );
+    }
+  }
+
+  const resp = await http.post(
+    `${documentURL}${referer.pid}/relations`,
+    newRelationsPayload
+  );
   resp.data = serializer.fromJSON(resp.data);
+
   return resp;
 };
 
-const deleteRelation = async (docPid, data) => {
-  const resp = await http.delete(`${documentURL}${docPid}/relations`, {
-    data: data,
-  });
+const deleteRelation = async (referer, related) => {
+  let deleteRequestPayload = {};
+  if (
+    related.relation_type === 'language' ||
+    related.relation_type === 'other' ||
+    related.relation_type === 'edition'
+  ) {
+    deleteRequestPayload = {
+      pid: related.pid,
+      pid_type: related.pid_type,
+      relation_type: related.relation_type,
+    };
+  } else {
+    deleteRequestPayload = {
+      parent_pid: related.pid,
+      parent_pid_type: related.pid_type,
+      child_pid: referer.metadata.pid,
+      child_pid_type: recordToPidType(referer),
+      relation_type: related.relation_type,
+    };
+  }
+  const resp = await http.delete(
+    `${documentURL}${referer.metadata.pid}/relations`,
+    {
+      data: [deleteRequestPayload],
```

Comment:
do we need to pass a list here?

Reply:
I didn't want to change the api on the backend until I will also redo the series relations